src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNet/SqlCommandIntegration.cs

```diff
@@ -218,11 +219,38 @@ public static class SqlCommandIntegration
                 throw;
             }
 
+            return AsyncHelper.InvokeGenericTaskDelegate(
```

Comment:
You could instead do like HttpMessageHandlerIntegration:

- Make the invoked async method return Task<object>
- Invoke it directly and use the TaskExtensions.Cast helper to convert the return type
- In the invoked async method, cast the return value of instrumentedMethod to Task to await it, then use `.GetProperty("Result").Value` to fetch the result

If was faster for HttpMessageHandlerIntegration, so I expect the same here. But we can do that in another PR if you prefer